nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/test/java/org/apache/nifi/parameter/aws/TestAwsSecretsManagerParameterProvider.java

```
@@ -0,0 +1,192 @@
+    );
+    final List<Parameter> otherSecretParameters = Arrays.asList(
+            parameter("paramG", "valueG"),
+            parameter("otherH", "valueOther")
+    );
+    final List<ParameterGroup> mockParameterGroups = Arrays.asList(
+            new ParameterGroup("MySecret", mySecretParameters),
+            new ParameterGroup("OtherSecret", otherSecretParameters)
+    );
+
+    private AwsSecretsManagerParameterProvider getParameterProvider() {
+        return spy(new AwsSecretsManagerParameterProvider());
+    }
+
+    private AWSSecretsManager mockSecretsManager(final List<ParameterGroup> mockGroup) {
+        final AWSSecretsManager secretsManager = mock(AWSSecretsManager.class);
+
+        final ListSecretsResult listSecretsResult = mock(ListSecretsResult.class);
+        final List<SecretListEntry> secretList = mockGroup.stream()
+                .map(group -> new SecretListEntry().withName(group.getGroupName()))
+                .collect(Collectors.toList());
+        when(listSecretsResult.getSecretList()).thenReturn(secretList);
+        when(secretsManager.listSecrets(any())).thenReturn(listSecretsResult);
+
+        mockGroup.forEach(group -> {
+            final String groupName = group.getGroupName();
+            final Map<String, String> keyValues = group.getParameters().stream().collect(Collectors.toMap(
+                    param -> param.getDescriptor().getName(),
+                    Parameter::getValue));
+            final String secretString;
+            try {
+                secretString = objectMapper.writeValueAsString(keyValues);
+                final GetSecretValueResult result = new GetSecretValueResult().withName(groupName).withSecretString(secretString);
+                when(secretsManager.getSecretValue(argThat(matchesGetSecretValueRequest(groupName))))
+                        .thenReturn(result);
+            } catch (final JsonProcessingException e) {
+                // Ignore
+            }
```

Instead of hiding the exception, it could be declared on the method and passed through to the test method

---

Since this is in a lambda, I'll just rethrow as an IllegalStateException